src/data_provider/src/sysInfo.cpp

```diff
@@ -50,41 +50,101 @@ nlohmann::json SysInfo::ports()
 #ifdef __cplusplus
 extern "C" {
 #endif
-void sysinfo_hardware(cJSON** js_result)
+int sysinfo_hardware(cJSON** js_result)
 {
-    SysInfo info;
-    const auto& hw          {info.hardware()};
-    *js_result = cJSON_Parse(hw.dump().c_str());
+    auto retVal { -1 };
+    try
+    {
+        SysInfo info;
+        const auto& hw          {info.hardware()};
+        *js_result = cJSON_Parse(hw.dump().c_str());
+        retVal = 0;
+    }
+    // LCOV_EXCL_START
+    catch(...)
+    {}
+    // LCOV_EXCL_STOP
+    return retVal;
 }
-void sysinfo_packages(cJSON** js_result)
+int sysinfo_packages(cJSON** js_result)
 {
-    SysInfo info;
-    const auto& packages    {info.packages()};
-    *js_result = cJSON_Parse(packages.dump().c_str());
+    auto retVal { -1 };
+    try
+    {
+        SysInfo info;
+        const auto& packages    {info.packages()};
+        *js_result = cJSON_Parse(packages.dump().c_str());
+        retVal = 0;
+    }
+    // LCOV_EXCL_START
+    catch(...)
+    {}
+    // LCOV_EXCL_STOP
+    return retVal;
 }
-void sysinfo_os(cJSON** js_result)
+int sysinfo_os(cJSON** js_result)
 {
-    SysInfo info;
-    const auto& os          {info.os()};
-    *js_result = cJSON_Parse(os.dump().c_str());
+    auto retVal { -1 };
+    try
+    {
+        SysInfo info;
+        const auto& os          {info.os()};
+        *js_result = cJSON_Parse(os.dump().c_str());
+        retVal = 0;
+    }
+    // LCOV_EXCL_START
+    catch(...)
+    {}
+    // LCOV_EXCL_STOP
+    return retVal;
 }
-void sysinfo_processes(cJSON** js_result)
+int sysinfo_processes(cJSON** js_result)
 {
-    SysInfo info;
-    const auto& processes   {info.processes()};
-    *js_result = cJSON_Parse(processes.dump().c_str());
+    auto retVal { -1 };
+    try
+    {
+        SysInfo info;
+        const auto& processes   {info.processes()};
+        *js_result = cJSON_Parse(processes.dump().c_str());
+        retVal = 0;
+    }
+    // LCOV_EXCL_START
+    catch(...)
+    {}
+    // LCOV_EXCL_STOP
+    return retVal;
 }
-void sysinfo_networks(cJSON** js_result)
+int sysinfo_networks(cJSON** js_result)
 {
-    SysInfo info;
-    const auto& networks    {info.networks()};
-    *js_result = cJSON_Parse(networks.dump().c_str());
+    auto retVal { -1 };
+    try
+    {
+        SysInfo info;
+        const auto& networks    {info.networks()};
+        *js_result = cJSON_Parse(networks.dump().c_str());
+        retVal = 0;
+    }
+    // LCOV_EXCL_START
+    catch(...)
+    {}
+    // LCOV_EXCL_STOP
+    return retVal;
 }
-void sysinfo_ports(cJSON** js_result)
+int sysinfo_ports(cJSON** js_result)
 {
-    SysInfo info;
-    const auto& ports       {info.ports()};
-    *js_result = cJSON_Parse(ports.dump().c_str());
+    auto retVal { -1 };
+    try
+    {
+        SysInfo info;
+        const auto& ports       {info.ports()};
+        *js_result = cJSON_Parse(ports.dump().c_str());
```

Comment:
Please add checks for the js_result pointers to avoid a crashes due to null ptr dereferencing.
Add a UT for the null ptr case.

Reply:
Will do